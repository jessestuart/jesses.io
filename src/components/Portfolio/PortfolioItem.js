/* @flow */
import React from 'react'
import { GitMerge } from 'react-feather'
import styled from 'styled-components'

import StyledLink from '../../components/StyledLink/StyledLink'
import colors from '../../utils/colors'

type Props = {
  link: string,
  title: string,
}

const GitMergeStyled = styled.div`
  transition: all 500ms;
  stroke: ${colors.primary.dark5};
  &:hover {
    stroke: ${colors.secondary.light8};
  }
`

const PortfolioItem = ({ link, title }: Props) => (
  <li className="f4 portfolio-item flex items-center mb2">
    <a href={link}>
      <GitMergeStyled>
        <GitMerge className="mr2" size="30px" color="inherit" />
      </GitMergeStyled>
    </a>
    <StyledLink className="fw7 f4" href={link}>
      {title}
    </StyledLink>
  </li>
)

export default PortfolioItem
