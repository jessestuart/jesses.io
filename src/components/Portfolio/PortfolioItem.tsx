import React from 'react'
import { Download } from 'react-feather'
import styled from 'styled-components'
import { StyledLink } from '../../components/StyledLink/StyledLink'
import { colors } from '../../utils/colors'

interface Props {
  link: string
  title: string
}

const GitMergeStyled = styled.div`
  transition: all 500ms;
  stroke: ${colors.secondary.dark5};
  &:hover {
    stroke: ${colors.secondary.main};
  }
`

const PortfolioItem = ({ link, title }: Props) => (
  <li className="f4 portfolio-item flex items-center mb2">
    <a href={link}>
      <GitMergeStyled>
        <Download className="mr2" size="27px" color="inherit" />
      </GitMergeStyled>
    </a>
    <StyledLink className="fw7 f4" href={link}>
      {title}
    </StyledLink>
  </li>
)

export default PortfolioItem
