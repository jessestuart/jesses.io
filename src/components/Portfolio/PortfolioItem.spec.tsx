import React from 'react'
import renderer from 'react-test-renderer'
import PortfolioItem from './PortfolioItem'

test('Render StyledLink component with default colors.', () => {
  const tree = renderer
    .create(<PortfolioItem link="/" title="Portfolio Item" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
