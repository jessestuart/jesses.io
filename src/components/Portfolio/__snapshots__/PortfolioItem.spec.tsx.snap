// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render StyledLink component with default colors. 1`] = `
<li
  className="f4 portfolio-item flex items-center mb2"
>
  <a
    href="/"
  >
    <div
      className="sc-bwzfXH keXXTO"
    >
      <svg
        className="mr2"
        fill="none"
        height="27px"
        stroke="inherit"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width="27px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
        />
        <polyline
          points="7 10 12 15 17 10"
        />
        <line
          x1="12"
          x2="12"
          y1="15"
          y2="3"
        />
      </svg>
    </div>
  </a>
  <a
    className="fw7 f4 sc-bdVaJa bXWskY"
    href="/true"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Portfolio Item
  </a>
</li>
`;
